scipy/stats/_hypotests.py

```diff
@@ -141,6 +141,211 @@ def epps_singleton_2samp(x, y, t=(0.4, 0.8)):
     return Epps_Singleton_2sampResult(w, p)
 
 
+PoissonMeansTestResult = make_dataclass('PoissonMeansTestResult',
+                                        ('statistic', 'pvalue'))
+
+
+def poisson_means_test(k1, n1, k2, n2, diff=0, alternative='two-sided'):
+    r"""
+    Calculates the poisson mean test, the "E-test", for the mean difference of
+    two samples that follow a Poisson distribution from descriptive statistics.
+
+    This is a two-sided test. The null hypothesis is that two independent
+    samples have identical average (expected) values.
+
+    Let :math:`X_{11},...,X_{1n_1}` and :math:`X_{21},...,X_{2n_2}` be
+    independent samples from distributions :math:`Poisson(\lambda_1)` and
+    :math:`Poisson(\lambda_2)`. It is well known that :math:`X_1`
+    and :math:`X_2` are independent:
+
+    .. math:: X_1 = \sum_{i=1}^{n_1} X_{1i} \sim Poisson(n_1\lambda_1)
+
+    .. math:: X_2 = \sum_{i=1}^{n_2} X_{2i} \sim Poisson(n_2\lambda_2)
+
+    Let `count1` and `count2` be the observed values of :math:`X_1` and
+    :math:`X_2`, respectively. The null hypothesis and alternative 
+    hypothesis under comparison are
+
+    .. math::
+       H_0: \lambda_1 = \lambda_2 + \mathtt{diff} \quad vs. \quad
+       H_a: \lambda_1 \ne \lambda_2 + \mathtt{diff}
+
+    for ``alternative=two-sided``, where :math:`\mathtt{diff} \ge 0`.
+
+    Parameters
+    ----------
+    k1 : int
+        Sample values of interest from sample 1.
+    n1: int
+        Sample size from sample 1.
+    k2 : int
+        Sample values of interest from sample 2.
+    n2: int
+        Sample size from sample 2.
+    diff : int or float, optional
+        The difference of mean between two samples under the null hypothesis
```

Comment:
Not sure if he saw the next line. Maybe specify the default as `diff : float, default=0`. 
I think the main point is that the null hypothesis above is "that two independent samples have identical average (expected) values." Then this implies that the null hypothesis is something else, something more general, but doesn't say it explicitly. If the null hypothesis above were rephrased in terms of the difference being a specified value, this could probably stay as is.